File: ANCIEN BACK/cible/domain/stock/services/structuring_service.py
from cible.domain.stock.interface.i_structuring_repository import IStructuringRepository
from cible.domain.stock.models.special_structuring_bean import SpecialStructuringBean
from cible.domain.stock.models.structuring_bean import StructuringBean


def get_all_structuring(repository: IStructuringRepository) -> list[StructuringBean]:
    return repository.get_all_structuring()


def get_all_special_structuring(
    repository: IStructuringRepository,
) -> list[SpecialStructuringBean]:
    return repository.get_all_special_structuring()


def delete_structuring(
    uuid: str, repository: IStructuringRepository
) -> list[StructuringBean]:
    return repository.delete_structuring(uuid)


def delete_special_structuring(
    uuid: str, repository: IStructuringRepository
) -> list[SpecialStructuringBean]:
    return repository.delete_special_structuring(uuid)


def create_structuring(
    repository: IStructuringRepository, structuring_bean: StructuringBean
) -> list[StructuringBean]:
    return repository.create_structuring(structuring_bean)


def create_special_structuring(
    repository: IStructuringRepository, special_structuring_bean: SpecialStructuringBean
) -> list[SpecialStructuringBean]:
    return repository.create_special_structuring(special_structuring_bean)


def create_structuring_from_excel_sheet(
    repository: IStructuringRepository, structuring_bean: StructuringBean
) -> None:
    return repository.create_structuring_from_excel_sheet(structuring_bean)


def create_special_structuring_from_excel_sheet(
    repository: IStructuringRepository, special_structuring_bean: SpecialStructuringBean
) -> None:
    return repository.create_special_structuring_from_excel_sheet(
        special_structuring_bean
    )


def add_additional_comment_to_structuring(
    repository: IStructuringRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_structuring(comment, object_uuid)


def add_additional_comment_to_special_structuring(
    repository: IStructuringRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_special_structuring(
        comment, object_uuid
    )

File: ANCIEN FRONT/services/stock/structuring.service.ts
import {fetcher} from "../utils/fetch-utils.service.ts";
import {BASE_API_CIBLE_HREF} from "../../constants.ts";
import SpecialStructuringModel from "../../core/domain/stock/SpecialStructuringModel.ts";
import StructuringModel from "../../core/domain/stock/StructuringModel.ts";

export function getAllStructuring(): Promise<StructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-structuring`, requestOptions);
}

export function getAllSpecialStructuring(): Promise<SpecialStructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-special-structuring`, requestOptions);
}

export function deleteStructuring(uuid: string): Promise<StructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/structuring/${uuid}`, requestOptions);
}

export function deleteSpecialStructuring(uuid: string): Promise<SpecialStructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/special-structuring/${uuid}`, requestOptions);
}

export function createStructuring(object: StructuringModel): Promise<StructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/structuring`, requestOptions);
}

export function createSpecialStructuring(object: SpecialStructuringModel): Promise<SpecialStructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/special-structuring`, requestOptions);
}

export function addAdditionalCommentToStructuring(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/structuring`, requestOptions);
}
export function addAdditionalCommentToSpecialStructuring(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/special-structuring`, requestOptions);
}